JetNews/app/src/main/java/com/example/jetnews/data/interests/impl/FakeInterestsRepository.kt

```diff
@@ -61,15 +68,44 @@ class FakeInterestsRepository : InterestsRepository {
         )
     }
 
-    override fun getTopics(callback: (Result<Map<String, List<String>>>) -> Unit) {
-        callback(Result.Success(topics))
+    // for now, keep the selections in memory
+    private val selectedTopics = MutableStateFlow(setOf<TopicSelection>())
+    private val selectedPeople = MutableStateFlow(setOf<String>())
+    private val selectedPublications = MutableStateFlow(setOf<String>())
+
+    override suspend fun getTopics(): Result<TopicsMap> {
+        return Result.Success(topics)
+    }
+
+    override suspend fun getPeople(): Result<List<String>> {
+        return Result.Success(people)
+    }
+
+    override suspend fun getPublications(): Result<List<String>> {
+        return Result.Success(publications)
     }
 
-    override fun getPeople(callback: (Result<List<String>>) -> Unit) {
-        callback(Result.Success(people))
+    override suspend fun toggleTopicSelection(topic: TopicSelection) {
+        val set = selectedTopics.value.toMutableSet()
```

Comment:
Nit: this (get, mutate, set) should really be synchronized (or forced onto one thread)